### ISSUE TITLE
LDtk bug

### ISSUE THREAD

**Issue by Tsunami014** [deepnight/ldtk#1051]
I was editing the enums in my level and then I pressed the 3 dots dropdown for one of them and then... this happened:

Stack:
```
LDtk version: 1.5.1-64bits
id.replace is not a function
TypeError
TypeError: id.replace is not a function
    at data_Project.cleanupIdentifier (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:35064:10)
    at data_Definitions.isEnumIdentifierUnique (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:32978:21)
    at file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:298433:40
    at data_Project.fixUniqueIdStr (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:35316:11)
    at i.fixValue (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:298432:34)
    at form_input_StringInput.onInputChange (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:128758:19)
    at HTMLInputElement.<anonymous> (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:128612:10)
    at HTMLInputElement.dispatch (C:\Users\TEMP.MAX-TSUNAMI014\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:43064)
    at v.handle (C:\Users\TEMP.MAX-TSUNAMI014\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:41048)
```
Processes:
```
#0 App
 |--- #9495 page.Editor
       |--- #9496 misc.FileWatcher
       |--- #9497 display.WorldRender
       |--- #9498 display.LevelRender
       |--- #9499 display.Camera
       |--- #9500 display.Rulers[--]
       |--- #9501 tool.SelectionTool[--]
       |--- #9502 tool.lt.DoNothing[--]
       |--- #9503 WorldTool
       |--- #9504 tool.PanView[--]
       |--- #9611 tool.lt.EntityTool[--]
       |--- #9804 ui.modal.panel.EditEnumDefs
       |--- #9970 display.EntityRender
       |--- #9971 display.EntityRender
       |--- #9972 display.EntityRender
```

Log:
```
[RENDER]       Rendering world bg...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[RENDER]       Rendering world bg...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[EVENT]        EnumDefValueRemoved
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[RENDER]       Rendering world bg...
[EVENT]        EnumDefAdded
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[EVENT]        EnumDefChanged
[RENDER]       Rendering world bg...
[RENDER]       Rendering world level fields data.Level#a2b563a0-66b0-11ec-9cd7-fd8a9b24800f "AutoLayers_advanced_demo"...
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        id.replace is not a function (TypeError)
[ERROR]        TypeError: id.replace is not a function
    at data_Project.cleanupIdentifier (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:35064:10)
    at data_Definitions.isEnumIdentifierUnique (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:32978:21)
    at file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:298433:40
    at data_Project.fixUniqueIdStr (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:35316:11)
    at i.fixValue (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:298432:34)
    at form_input_StringInput.onInputChange (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:128758:19)
    at HTMLInputElement.<anonymous> (file:///C:/Users/TEMP.MAX-TSUNAMI014/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:128612:10)
    at HTMLInputElement.dispatch (C:\Users\TEMP.MAX-TSUNAMI014\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:43064)
    at v.handle (C:\Users\TEMP.MAX-TSUNAMI014\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:41048)

[GENERAL]      
#0 App
 |--- #9495 page.Editor
       |--- #9496 misc.FileWatcher
       |--- #9497 display.WorldRender
       |--- #9498 display.LevelRender
       |--- #9499 display.Camera
       |--- #9500 display.Rulers[--]
       |--- #9501 tool.SelectionTool[--]
       |--- #9502 tool.lt.DoNothing[--]
       |--- #9503 WorldTool
       |--- #9504 tool.PanView[--]
       |--- #9611 tool.lt.EntityTool[--]
       |--- #9804 ui.modal.panel.EditEnumDefs
       |--- #9970 display.EntityRender
       |--- #9971 display.EntityRender
       |--- #9972 display.EntityRender
 |--- #9973 page.CrashReport

[FILE]         Cleared all file watches
```

**Comment by deepnight** [comment]
Thanks for reporting!

**Comment by deepnight** [comment]
Do you mean those dots?

![image](https://github.com/deepnight/ldtk/assets/1671695/0d85cc60-bb65-4b82-b847-8f62361dfc65)

**Comment by deepnight** [comment]
Fixed in 1.5.3
